src/engine/renderer/tr_shade.cpp

```diff
@@ -598,6 +598,12 @@ void SetNormalScale( shaderStage_t *pStage, vec3_t normalScale)
 			normalScale[ 1 ] *= pStage->normalScale[ 1 ];
 			normalScale[ 2 ] *= pStage->normalScale[ 2 ];
 		}
+
+		// never multiply normal Z by zero, it's wrong
+		if ( r_normalScale->integer )
```

Comment:
I think this should be a float, not integer.

 I'd drop the zero test also. I imagine the main purpose of this variable is debugging, to make the normals stand out more so you can check them. So 0 would also be a legitimate value, to test that things become flat as expected (although it wouldn't work now I guess, due to testing the z component in glsl).

Reply:
Hmm, right, I'll see if it's a float.

In fact, I have a strong feeling some of those computes are wrong. At this point I'm just moving all of them at the same place, keeping the legacy behavior. Then once code is clean, it will be time to ask if it makes sense or not. Initially we had a `normalScale` expression that was expected to operate on X and Y,  and a `r_normalScale` cvar that was expected to operate on Z… I'll investigate on those things later. My first priority is to have a cleaner code so we find easily where things are done so we can question if it is the right way to do things or not and fix it easily.

Reply:
float fixed.